internal/controller/finrestore_controller_test.go

```diff
@@ -486,6 +545,150 @@ var _ = Describe("FinRestore Controller Reconcile Test", Ordered, func() {
+			//
+			// Arrange:
+			//   - Create a FinBackup with ChecksumMismatched=True.
+			//   - Create a FinRestore with allowChecksumMismatched=true.
+			//
+			// Act:
+			//   Run reconciliation to create and complete the restore job.
+			//
+			// Assert:
+			//   - ENABLE_CHECKSUM_VERIFY is set to false in the restore Job.
+			//   - FinRestore becomes ReadyToUse after job completion.
+
+			// Arrange
+			By("creating FinBackup with ChecksumMismatched=True")
+			finbackup := CreateFinBackupStoredAndVerified(ctx, k8sClient, namespace, utils.GetUniqueName("test-fin-backup"), pvc, 1, utils.GetUniqueName("test-node"))
+			meta.SetStatusCondition(&finbackup.Status.Conditions, metav1.Condition{
+				Type:    finv1.BackupConditionChecksumMismatched,
+				Status:  metav1.ConditionTrue,
+				Reason:  "ChecksumMismatch",
+				Message: "Checksum corruption detected",
+			})
+			Expect(k8sClient.Status().Update(ctx, finbackup)).Should(Succeed())
+
+			By("creating FinRestore with allowChecksumMismatched=true")
+			finrestore := NewFinRestore(
+				namespace,
+				utils.GetUniqueName("test-restore"),
+				finbackup.Name,
+				utils.GetUniqueName("restore-pvc"),
+				namespace,
+			)
+			finrestore.Spec.AllowChecksumMismatched = true
+			Expect(k8sClient.Create(ctx, finrestore)).Should(Succeed())
+
+			// Act
+			By("reconciling the FinRestore to create restore job")
+			_, err := reconciler.Reconcile(ctx, ctrl.Request{NamespacedName: client.ObjectKeyFromObject(finrestore)})
+			Expect(err).ShouldNot(HaveOccurred())
+
+			createAndBindRestorePV(ctx, finrestore)
+
+			By("reconciling again after PVC is bound")
+			_, err = reconciler.Reconcile(ctx, ctrl.Request{NamespacedName: client.ObjectKeyFromObject(finrestore)})
+			Expect(err).ShouldNot(HaveOccurred())
+
+			// Assert
+			By("checking that ENABLE_CHECKSUM_VERIFY is false in the restore Job")
+			expectRestoreJobChecksumVerify(ctx, finrestore, "false")
+
+			By("making the restore job complete")
+			restoreJobKey := client.ObjectKey{Namespace: namespace, Name: restoreJobName(finrestore)}
+			Eventually(func(g Gomega, ctx SpecContext) {
+				var restoreJob batchv1.Job
+				g.Expect(k8sClient.Get(ctx, restoreJobKey, &restoreJob)).To(Succeed())
+				makeJobSucceeded(&restoreJob)
+				g.Expect(k8sClient.Status().Update(ctx, &restoreJob)).To(Succeed())
+			}, "5s", "1s").WithContext(ctx).Should(Succeed())
+
+			By("reconciling to process job completion")
+			_, err = reconciler.Reconcile(ctx, ctrl.Request{NamespacedName: client.ObjectKeyFromObject(finrestore)})
+			Expect(err).ShouldNot(HaveOccurred())
+
+			By("checking that FinRestore becomes ReadyToUse")
+			Eventually(func(g Gomega) {
+				var updated finv1.FinRestore
+				err := k8sClient.Get(ctx, client.ObjectKeyFromObject(finrestore), &updated)
+				g.Expect(err).ShouldNot(HaveOccurred())
+				g.Expect(updated.IsReady()).Should(BeTrue())
+			}, "5s", "1s").Should(Succeed())
```

Comment:
This test seems excessive; it may be enough to check that the restoration PVC is created.